lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,128 @@
+
+        logger.debug("YAML HASH BELOW:")
+        logger.debug(file_hash)
+        logger.debug("PLATFORM_DATA BELOW:")
+        logger.debug(file_hash[:platform_data])
+        return sha_yaml_folder_url, file_hash[:platform_data]
+      end
+
+      # get the build artifact URL for a host
+      #
+      # @param [Host] host Host to get artifact URL for
+      # @param [Hash] build_details Details of the build in a hash
+      # @param [String] build_url URL to the build
+      #
+      # @return [String] URL to the build artifact
+      def hosts_artifact_url(host, build_details, build_url)
+        packaging_platform = host[:packaging_platform]
+        if packaging_platform.nil?
+          message = <<-EOF
+            :packaging_platform not provided for host '#{host}', platform '#{host[:platform]}'
+            :packaging_platform should be the platform-specific key from this list:
+              #{ build_details.keys }
+          EOF
+          fail_test( message )
+        end
+
+        logger.debug("PLATFORM SPECIFIC INFO for #{host} (packaging name '#{packaging_platform}'):")
+        packaging_data = build_details[packaging_platform]
+        logger.debug("- #{ packaging_data }, isnil? #{ packaging_data.nil? }")
+        artifact_buildserver_path = packaging_data[:artifact]
+        fail_test('no artifact_buildserver_path found') if artifact_buildserver_path.nil?
+
+        artifact_url = "#{build_url}/#{artifact_buildserver_path}"
+        artifact_url_correct = link_exists?( artifact_url )
+        logger.debug("- artifact url: '#{artifact_url}'. Exists? #{artifact_url_correct}")
+        fail_test('artifact url built incorrectly') if !artifact_url_correct
+        artifact_url
+      end
+
+      # install build artifact on the given host
+      #
+      # @param [Host] host Host to install artifact on
+      # @param [String] project_name Name of project for artifact. Needed for OSX installs
+      # @param [String] artifact_url URL of the project install artifact
+      #
+      # @return nil
+      def install_artifact_on(host, artifact_url, project_name)
+        variant, _, _, _ = host[:platform].to_array
+        onhost_package_file = nil
+        if variant == 'solaris'
+          artifact_filename = File.basename( artifact_url )
+          artifact_folder = File.dirname( artifact_url )
+          fetch_http_file( artifact_folder, artifact_filename, '.' )
+          scp_to host, artifact_filename, '/tmp'
+          onhost_package_file = "/tmp/#{ artifact_filename }"
+        elsif variant == 'debian'
+          onhost_package_file = '/tmp/puppet.deb'
+          on host, "wget -O #{ onhost_package_file } #{ artifact_url }"
+        elsif variant == 'osx'
+          on host, "curl -O #{ artifact_url }"
+          onhost_package_file = "#{ project_name }*"
+        end
+
+        if onhost_package_file
+          host.install_local_package( onhost_package_file )
+        else
+          host.install_package( artifact_url )
```

Comment:
Hey all! We're trying to have a single source of truth for the supported platforms. Anything in github should be based on the confluence page, but should not be considered a final source of truth. For the final source of truth, consult https://confluence.puppetlabs.com/display/PROD/Currently+Supported+OS+Platforms. However, it does look like it has out of date information for yakkety :(

A lot of this decision not to ship Yakkey comes from the following email conversation between Ruth, @joshcooper and @haus 

>>> (Note: probably only the platform team cares about this, but I don't know what the appropriate email list is for that that definitely includes the people who care about this and I already spent 5 minutes trying to figure it out before giving up.)

>>>The plaftorm team has done some work to get puppet-agent and puppet server packages for Yakkety (Ubuntu 16.10) for Puppet 5... but there are still bugs to sort out causing redness in pipelines.

>>Curious if the bugs are specific to Yakkety-only, or will we hit the same issues as soon as we try to add support for Zesty 17.04 in https://tickets.puppetlabs.com/browse/PA-873?

>Yakkety has the weird puppet-agent package that isn't from us, while Zesty does not, fwiw.
 
>>>Given that the current date for GA for Puppet 5 is end of June and Yakkety goes EOL in July, it seems like rather than trying to spend more time getting this to work, our time could be better spent elsewhere.

>>>I talked with Eric0 briefly and he was on board with not shipping Yakkety packages for Puppet platform 5, but I wasn't sure what further things (confluence pages??) need updating, and also wanted to make sure this discussion was widely circulated.